libsolidity/interface/CompilerStack.cpp

```diff
@@ -1540,6 +1540,7 @@ string CompilerStack::createMetadata(Contract const& _contract, bool _forIR) con
 			details["yulDetails"] = Json::objectValue;
 			details["yulDetails"]["stackAllocation"] = m_optimiserSettings.optimizeStackAllocation;
 			details["yulDetails"]["optimizerSteps"] = m_optimiserSettings.yulOptimiserSteps;
+			details["yulDetails"]["optimizerCleanupSteps"] = m_optimiserSettings.yulOptimiserCleanupSteps;
```

Comment:
Yeah, damn, I really shouldn't have missed this earlier, shame on me :-)...

But wouldn't it be better to serialize this into ``optimizerSteps`` with a colon, though? That way consuming the metadata would stay compatible, wouldn't it?
If we add a new field, this needs specific support.

Reply:
As for some more context: given some deployed bytecode, tools like sourcify fetch the metadata for the contract, which contains source references and compiler settings - then those sources are fetched and recompiled with the respective settings and it's confirmed that the resulting bytecode is identical to the one you started from (there's some additional pains involved in the process, but basically).

So if we add a new field here, such tools would need to become aware of it to correctly feed it back to the compiler for recompilation. If we just serialize it into ``optimizerSteps`` with the delimiter, though, the current logic of passing it to the compiler should just work - that's why I'd say that'd be preferable.

Reply:
Should be trivial, i.e. what you're suggesting is?
```c++
details["yulDetails"]["optimizerSteps"] = m_optimiserSettings.yulOptimiserSteps + ":" + m_optimiserSettings.yulOptimiserCleanupSteps;
```
Now, if I change it, will @cameel be OK with that?

Reply:
That's what I'm suggesting, yeah :-).

Reply:
Sure thing

Reply:
Changed + added boost tests

Reply:
Yeah, looks good.